service/handlers_models_test.go

```diff
@@ -29,133 +29,195 @@ import (
-	}
-	if result.Success {
-		t.Error("Expected error, got success response")
-	}
-	if result.ErrorCode != "error-auth" {
-		t.Error("Expected error-auth code")
-	}
+func (s *ModelsSuite) TestModelsHandlerWithoutPermissions(c *check.C) {
+	w, err := s.sendGETRequest("/v1/models", datastore.Invalid)
+	c.Assert(err, check.IsNil)
+	c.Assert(w.Code, check.Equals, 200)
+
+	result, err := s.parseModelsResponse(w)
+	c.Assert(result.Success, check.Equals, false)
+	c.Assert(result.ErrorCode, check.Equals, "error-auth")
 }
 
-func TestModelsHandlerWithError(t *testing.T) {
+func (s *ModelsSuite) TestModelsHandlerWithError(c *check.C) {
+	datastore.Environ.Config.EnableUserAuth = false
+	datastore.Environ.DB = &datastore.ErrorMockDB{}
 
-	// Mock the database
-	config := config.Settings{JwtSecret: "SomeTestSecretValue"}
-	datastore.Environ = &datastore.Env{DB: &datastore.ErrorMockDB{}, Config: config}
+	w, err := s.sendGETRequest("/v1/models", datastore.Invalid)
+	c.Assert(err, check.IsNil)
+	c.Assert(w.Code, check.Equals, 400)
 
-	w := httptest.NewRecorder()
-	r, _ := http.NewRequest("GET", "/v1/models", nil)
-	http.HandlerFunc(ModelsHandler).ServeHTTP(w, r)
+	result, err := s.parseModelsResponse(w)
+	c.Assert(result.Success, check.Equals, false)
+}
 
-	// Check the JSON response
-	result := ModelsResponse{}
-	err := json.NewDecoder(w.Body).Decode(&result)
-	if err != nil {
-		t.Errorf("Error decoding the models response: %v", err)
-	}
-	if result.Success {
-		t.Error("Expected error, got success response")
-	}
+func (s *ModelsSuite) TestModelGetHandler(c *check.C) {
+	w, err := s.sendGETRequest("/v1/models/1", datastore.Admin)
+	c.Assert(err, check.IsNil)
+	c.Assert(w.Code, check.Equals, 200)
 
+	result, err := s.parseModelResponse(w)
+	c.Assert(err, check.IsNil)
+	c.Assert(result.Success, check.Equals, true)
+	c.Assert(result.Model.ID, check.Equals, 1)
+	c.Assert(result.Model.Name, check.Equals, "alder")
 }
 
-func TestModelGetHandler(t *testing.T) {
+func (s *ModelsSuite) TestModelAssertionHeadersHandler(c *check.C) {
+	d := datastore.ModelAssertion{
+		ModelID: 1, KeypairID: 1,
+		Series: 16, Architecture: "amd64", Revision: 1,
+		Gadget: "mygadget", Kernel: "mykernel", Store: "ubuntu",
+	}
+	data, _ := json.Marshal(d)
 
-	// Mock the database
-	config := config.Settings{JwtSecret: "SomeTestSecretValue"}
-	datastore.Environ = &datastore.Env{DB: &datastore.MockDB{}, Config: config}
+	w, err := s.sendPOSTRequest("/v1/models/assertion", bytes.NewReader(data), datastore.Admin)
+	c.Assert(err, check.IsNil)
+	c.Assert(w.Code, check.Equals, 200)
 
-	result, _ := sendRequest(t, "GET", "/v1/models/1", nil)
+	result, err := s.parseModelResponse(w)
+	c.Assert(err, check.IsNil)
+	c.Assert(result.Success, check.Equals, true)
+}
 
-	if result.Model.ID != 1 {
-		t.Errorf("Expected model with ID 1, got %d", result.Model.ID)
+func (s *ModelsSuite) TestModelAssertionHeadersHandlerWithoutPermissions(c *check.C) {
+	d := datastore.ModelAssertion{
```

Comment:
It's fine as is, but just in case, do you know that you can define whatever fields you need into ModelSuite struct and reference them in the different tests?